packages/frontend/src/components/common/FieldSelect/index.tsx

```diff
@@ -86,13 +91,47 @@ const FieldSelect = <T extends Item = Item>({
     onClosed,
     inactiveItemIds = [],
     hasGrouping = false,
+    tableName,
     ...rest
 }: FieldSelectProps<T>): JSX.Element => {
     const sortedItems = useMemo(() => {
-        return items.sort((a, b) =>
-            getLabel(a, hasGrouping).localeCompare(getLabel(b, hasGrouping)),
-        );
-    }, [items, hasGrouping]);
+        // Give priority to items in the table === tableName
```

Comment:
I can move this to a util function 

I also tried to mimic the logic in `ExploreTree/index`
```
const tableTrees = useMemo(() => {
        return Object.values(explore.tables)
            .sort((tableA, tableB) => {
                if (tableA.name === explore.baseTable) return -1;
                if (tableB.name === explore.baseTable) return 1;
                // Sorting explores by label
                return tableA.label.localeCompare(tableB.label);
            })
            .filter(
                (table) =>
                    !(isSearching && !searchHasResults(table)) && !table.hidden,
            );
    }, [explore, searchHasResults, isSearching]);
```

But these are diff structures.